File: TixClick_FrontEnd/src/interface/company/Company.ts
export interface Company {
  companyId: number;
  companyName: string;
  codeTax: string;
  bankingName: string;
  bankingCode: string;
  email: string;
  ownerCard: string;
  nationalId: string;
  logoURL: string;
  address: string;
  description: string;
  status: string;
  representativeId: number;
}

export type CompanyStatus = "ACTIVE" | "PENDING" | "REJECTED";

interface CompanyList extends Company {
  subRole: string;
}

export interface CompanyListResponse {
  myCompany: CompanyList;
  listCompany: CompanyList[];
}

File: TixClick_BackEnd/src/main/java/com/pse/tixclick/payload/entity/company/Company.java
package com.pse.tixclick.payload.entity.company;

import com.pse.tixclick.payload.entity.Account;
import com.pse.tixclick.payload.entity.entity_enum.ECompanyStatus;
import com.pse.tixclick.payload.entity.entity_enum.EStatus;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
@Builder
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int companyId;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String companyName;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String codeTax;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String bankingName;

    @Column(columnDefinition = "NVARCHAR(255)")
    private String ownerCard;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String bankingCode;

    @Column(columnDefinition = "NVARCHAR(255)", nullable = false)
    private String nationalId;

    @Column(columnDefinition = "NVARCHAR(255)")
    private String address;

    @Column(columnDefinition = "NVARCHAR(255)")
    private String email;

    @Column
    private String logoURL;

    @Column(columnDefinition = "NVARCHAR(255)")
    private String description;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    ECompanyStatus status;

    @OneToOne
    @JoinColumn(name = "representative_id", nullable = false)
    private Account representativeId;

}
